src/pygama/raw/data_trimmer/data_trimmer.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import copy
+import json
+
+import pygama.lgdo as lgdo
+from pygama.dsp.processing_chain import build_processing_chain as bpc
+from pygama.lgdo import Array, Struct, Table
+
+
+def data_trimmer(lgdo_table: Table | Struct, dsp_config: str | dict) -> None:
+    """
+    Takes in an lgdo table that contains waveforms, performs user specified dsp on the table, and then updates the table in place
+
+    Parameters
+    ----------
+    lgdo_table
+        An lgdo Table or Struct that must contain waveforms so that the dsp can work!
```

Comment:
```suggestion
        An :class:`~.lgdo.table.Table` or :class:`~.lgdo.struct.Struct` that must contain waveforms so that the DSP
        can work!
```